packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx

```diff
@@ -292,195 +301,193 @@ const BulkActionModal = ({
-//           type: 'warning',
-//           message: formatAPIError(error),
-//         });
-//       },
-//     }
-//   );
-
-//   const handleConfirmUnpublishAllData = async () => {
-//     await bulkUnpublishMutation.mutateAsync({ ids });
-//   };
-
-//   const showUnpublishButton =
-//     hasPublishPermission && selectedEntriesObjects.some((entry) => entry.publishedAt);
-
-//   if (!showUnpublishButton) return null;
-
-//   return {
-//     actionType: 'unpublish',
-//     variant: 'tertiary',
-//     label: formatMessage({ id: 'app.utils.unpublish', defaultMessage: 'Unpublish' }),
-//     dialog: {
-//       type: 'dialog',
-//       title: formatMessage({
-//         id: 'app.components.ConfirmDialog.title',
-//         defaultMessage: 'Confirmation',
-//       }),
-//       content: (
-//         <Flex direction="column" alignItems="stretch" gap={2}>
-//           <Typography id="confirm-description" textAlign="center">
-//             {formatMessage({
-//               id: 'popUpWarning.bodyMessage.contentType.unpublish.all',
-//               defaultMessage: 'Are you sure you want to unpublish these entries?',
-//             })}
-//           </Typography>
-//           {hasI18nEnabled && (
-//             <Box textAlign="center" padding={3}>
-//               <Typography textColor="danger500">
-//                 {formatMessage(
-//                   {
-//                     id: getTranslation('Settings.list.actions.unpublishAdditionalInfos'),
-//                     defaultMessage:
-//                       'This will unpublish the active locale versions <em>(from Internationalization)</em>',
-//                   },
-//                   {
-//                     em: Emphasis,
-//                   }
-//                 )}
-//               </Typography>
-//             </Box>
-//           )}
-//         </Flex>
-//       ),
-//       confirmButton: formatMessage({
-//         id: 'app.utils.unpublish',
-//         defaultMessage: 'Unpublish',
-//       }),
-//       onConfirm: handleConfirmUnpublishAllData,
-//     },
-//   };
-// };
+const BULK_DELETE_OPTIONS = {
+  SELECTED_LOCALE: 'selected_Locale',
+  ALL_LOCALES: 'all_Locales',
+};
+
+const DeleteAction: BulkActionComponent = ({ documentIds, model }) => {
+  const { formatMessage } = useIntl();
+  const { selectRow } = useTable('deleteAction', (state) => state);
+  const [{ query }] = useQueryParams<{ plugins?: { i18n?: { locale?: string } } }>();
+  const currentLocale = query.plugins?.i18n?.locale || 'en';
+  // const { data: locales = [] } = useGetLocalesQuery(); TODO: check if can import this query to get locale name
+  const hasDeletePermission = useDocumentRBAC('deleteAction', (state) => state.canDelete);
+  const { deleteMany: bulkdeleteAction } = useDocumentActions();
+  const [isDeleteAllLocale, setIsDeleteAllLocale] = React.useState(false);
+
+  const handleConfirmBulkDelete = async () => {
+    const data = await bulkdeleteAction({
```

Comment:
lets call this `res` like we do with most other actions since it can contain `data`.